### ISSUE TITLE
Reconnect on disconnection

### ISSUE THREAD

**Issue by jimtng** [billchurch/bhyve-mqtt#7]
When the websocket got disconnected, bhyve-mqtt failed and exited. Instead, please handle the disconnection and try to reconnect.

**Comment by billchurch** [comment]
Could you provide a log or some more detail behind this? I tried to simulate the crash and I wasn't able to reproduce... 

I get a `Orbit Error: Error: read ECONNRESET` but not an exit. Obviously this condition needs to be handled but I'm not seeing the behavior you report. 

```
2020-01-24T12:50:58.501Z - parseMessage topic: bhyve/device/refresh
2020-01-24T12:50:58.502Z - refresh
2020-01-24T12:50:58.829Z - Orbit Error: Error: read ECONNRESET
```
It's worth noting that if you publish `bhyve/device/refresh` it will reconnect, at least it is in my case. 
```
2020-01-24T12:51:32.851Z - parseMessage topic: bhyve/device/refresh
2020-01-24T12:51:32.851Z - refresh
2020-01-24T12:51:33.280Z - subscribe topic: bhyve/device/refresh
2020-01-24T12:51:33.280Z - granted: []
2020-01-24T12:51:33.280Z - devices: {"last_connected_at":"2020-01-22T13:11:01.668Z",...
```

**Comment by billchurch** [comment]
Are you possibly talking about an mqtt client disconnection? That *will* trigger a process exit.

**Comment by tltgt** [comment]
```
2020-01-24T09:15:02.064Z - websocket sending ping
/xxxx/bhyve-mqtt/app/node_modules/ws/lib/websocket.js:335
      throw err;
      ^

Error: WebSocket is not open: readyState 0 (CONNECTING)
    at WebSocket.send (/xxxx/bhyve-mqtt/app/node_modules/ws/lib/websocket.js:329:19)
    at Timeout.sendPing [as _onTimeout] (/xxxx/bhyve-mqtt/app/orbit.js:132:18)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bhyve-mqtt@0.1.0 start: `node -r dotenv/config app.js dotenv_config_path=./.env`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bhyve-mqtt@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /xxxx/.npm/_logs/2020-01-24T09_15_02_078Z-debug.log
```

**Comment by jimtng** [comment]
```
2020-01-24T09:15:02.064Z - websocket sending ping
/xxxx/bhyve-mqtt/app/node_modules/ws/lib/websocket.js:335
      throw err;
      ^

Error: WebSocket is not open: readyState 0 (CONNECTING)
    at WebSocket.send (/xxxx/bhyve-mqtt/app/node_modules/ws/lib/websocket.js:329:19)
    at Timeout.sendPing [as _onTimeout] (/xxxx/bhyve-mqtt/app/orbit.js:132:18)
    at ontimeout (timers.js:436:11)
    at tryOnTimeout (timers.js:300:5)
    at listOnTimeout (timers.js:263:5)
    at Timer.processTimers (timers.js:223:10)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! bhyve-mqtt@0.1.0 start: `node -r dotenv/config app.js dotenv_config_path=./.env`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bhyve-mqtt@0.1.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /xxxx/.npm/_logs/2020-01-24T09_15_02_078Z-debug.log
```

**Comment by dragon299** [comment]
I have the same issue and would like to see that fixed

**Comment by billchurch** [comment]
I'm not quite dead yet. Looking at this again.

**Comment by jimtng** [comment]
I've  replaced my orbit bhyve with a wifi based valve that I've flashed with Tasmota.

**Comment by billchurch** [comment]
Cool, what did you get?

**Comment by jimtng** [comment]
This is the one I got: https://www.aliexpress.com/item/32985685144.html

They do need a USB power source though, but since one of the problems I had with orbit bhyve was that its batteries don't last long (I was using nimh rechargeables, e.g. eneloop - they last for 2-3 months on average), the fact that these ones are plugged in means I don't have to worry about flat batteries. 

I set the watering schedule using Tasmota's built in timers. Then using openHAB, I send a command to Tasmota to disable the timers (for up to 24 hours) when rain is predicted in the next 24 hours. 

Tasmota will reactivate its timers by itself once the "rain delay" has expired, so in the worst case if openhab is down, it will just water as per schedule regardless of rain.
